portfolio/src/main/webapp/indexScript.js

```diff
@@ -0,0 +1,58 @@
+window.addEventListener('DOMContentLoaded', (event) => {
+    getBlogs();
+
+    const selectElement = document.getElementById('cat-select');     
+    selectElement.addEventListener('change', function() {
+      getBlogs();
+     });
+});
+
+function getBlogs() {
+    fetch('/post').then(response => response.json()).then((posts) => {
+      const blogListElement = document.getElementById('blog-posts');     
+      const selectedElement = document.getElementById('cat-select');
+      blogListElement.innerHTML = '';  
+
+      for (post of posts) {
+        if (selectedElement.value === 'All') {
+            blogListElement.appendChild(createBlogElement(post));
+        }
+        else if (post.category === selectedElement.value) {
+            blogListElement.appendChild(createBlogElement(post));
+        }
```

Comment:
Both your if and else if cases do the same thing, I'd suggest combining the two checks:
`if (selectedElement.value === 'All' || post.category === selectedElement.value)`